test/e2e/operator/nmstate_install_test.go

```diff
@@ -0,0 +1,158 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	v1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1beta1"
+	"github.com/nmstate/kubernetes-nmstate/test/e2e/daemonset"
+	"github.com/nmstate/kubernetes-nmstate/test/e2e/deployment"
+)
+
+var _ = Describe("NMState operator", func() {
+	Context("when installed for the first time", func() {
+		BeforeEach(func() {
+			installDefaultNMState()
+		})
+		It("should deploy daemonset and webhook deployment", func() {
+			daemonset.DaemonSet(getDefaultNMStateHandlerKey()).ReadyConditionEventually().Should(BeTrue())
+			deployment.Deployment(getDefaultNMStateWebhookKey()).ReadyConditionEventually().Should(BeTrue())
+		})
+	})
+	Context("when NMState is installed", func() {
+		BeforeEach(func() {
+			ds, err := getDefaultNMStateHandlerList()
+			Expect(err).ToNot(HaveOccurred(), "List daemon sets in namespace nmstate succeeds")
+			Expect(ds.Items).To(HaveLen(1), "and has only one item")
+		})
+		Context("and the CR is created with a wrong name", func() {
+			var nmstate = getNMStateWrongName()
```

Comment:
Just use `getNMStateWrongName` at `BeforeEach` and `AfterEach` so it's clear there what is happening